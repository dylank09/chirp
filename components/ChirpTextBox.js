import React from "react";
import { StyleSheet, TextInput } from "react-native";

import { theme } from "../assets/Theme";

export default function ChirpTextBox({
  onChangeText,
  value,
  placeholder,
  password,
  allowMultiline,
}) {
  return (
    <TextInput
      style={styles.textBox}
      value={value}
      onChangeText={onChangeText ? onChangeText : null}
      secureTextEntry={password ? true : false}
      selectTextOnFocus={password ? true : false}
      selectionColor={theme.colors.primary}
      multiline={allowMultiline ? allowMultiline : false}
      blurOnSubmit={true}
      placeholder={placeholder ? placeholder : ""}
      placeholderTextColor={theme.colors.placeholderColor}
    ></TextInput>
  );
}

const styles = StyleSheet.create({
  textBox: {
    padding: 5,
    width: theme.dimensions.authWidth,
    color: theme.colors.text,
    fontSize: theme.dimensions.standardFontSize + 1,
    borderBottomColor: theme.colors.text,
    borderBottomWidth: 1,
    marginTop: 7,
    marginBottom: 5,
  },
});
