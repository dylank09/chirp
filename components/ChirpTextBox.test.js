import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";

import ChirpTextBox from "./ChirpTextBox";

describe("<ChirpTextBox />", () => {
  const tree = renderer.create(<ChirpTextBox />).toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("text box is present", () => {
    const { getByTestId } = render(<ChirpTextBox value="Testing" />);
    const foundButton = getByTestId("chirpTextBox");
    expect(foundButton).toBeTruthy();
  });
});
