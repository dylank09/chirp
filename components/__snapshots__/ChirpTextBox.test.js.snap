// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChirpTextBox /> renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  blurOnSubmit={true}
  multiline={false}
  onChangeText={null}
  placeholder=""
  placeholderTextColor="rgb(158, 158, 158)"
  rejectResponderTermination={true}
  secureTextEntry={false}
  selectTextOnFocus={false}
  selectionColor="rgb(193, 73, 83)"
  style={
    Array [
      Object {
        "borderBottomColor": "rgb(255, 255, 255)",
        "borderBottomWidth": 1,
        "color": "rgb(255, 255, 255)",
        "width": "82%",
      },
      Object {
        "fontSize": 15,
        "marginBottom": 4,
        "marginTop": 7,
        "padding": 5,
      },
    ]
  }
  testID="chirpTextBox"
  underlineColorAndroid="transparent"
/>
`;
